src/host/os_linkdir.c

```diff
@@ -0,0 +1,61 @@
+/**
+ * \file os_linkdir.c
+ * \brief Creates a symbolic link to a directory.
+ * \author Copyright (c) 2024 Jess Perkins and the Premake project
+ */
+
+#include <sys/stat.h>
+#include "premake.h"
+
+int do_linkdir(lua_State* L, const char* src, const char* dst)
+{
+#if PLATFORM_WINDOWS
+    // Prepend the drive letter if a relative path is given
+    char dstPath[MAX_PATH];
+	char srcPath[MAX_PATH];
+
+	do_normalize(L, srcPath, src);
+	do_normalize(L, dstPath, dst);
+	do_translate(dstPath, '\\');
+	do_translate(srcPath, '\\');
+
+	// If the source path is relative, prepend the current working directory
+	if (!do_isabsolute(src))
+	{
+		char cwd[MAX_PATH];
+		do_getcwd(cwd, MAX_PATH);
+		do_translate(cwd, '\\');
+
+		char relSrcPath[MAX_PATH];
+		snprintf(relSrcPath, MAX_PATH, "%c:%s", cwd[0], srcPath);
+
+		BOOLEAN res = CreateSymbolicLinkA(dstPath, relSrcPath, SYMBOLIC_LINK_FLAG_DIRECTORY | SYMBOLIC_LINK_FLAG_ALLOW_UNPRIVILEGED_CREATE);
```

Comment:
Would it work with path with non-ascii characters? I don't know if/how we support unicode...

Reply:
The functions I rely on (from premake) don't directly have `wchar_t` support. Maybe that's something worth investigating.